source/extensions/filters/common/lua/lua.h

```diff
@@ -60,6 +61,31 @@ namespace Lua {
  */
 #define DECLARE_LUA_CLOSURE(Class, Name) DECLARE_LUA_FUNCTION_EX(Class, Name, lua_upvalueindex(1))
 
+/**
+ * Calculate the maximum space needed to be aligned.
+ */
+template <typename T> constexpr size_t maximumSpaceNeededToAlign() {
+  return sizeof(T) + alignof(T) - 1;
```

Comment:
Can you add some more comments about what this math is doing?

Reply:
done